### PR TITLE
refactor(worker): locked task middleware run

### PR DESCRIPTION
**Pull request by Guikingone**
| Q                | A
| ---------------- | ---
| PHP version?     | 7.4
| Bundle version?  | next (as patch)
| New feature?     | no
| Bug fix?         | yes

### CODE REVIEW

File: src/Worker/Worker.php [hunk 1]
```diff
@@ -132,12 +132,14 @@ public function execute(array $options = [], TaskInterface ...$tasks): void
                     try {
                         $this->middlewareStack->runPreExecutionMiddleware($task);
 
-                        if ($lockedTask->acquire() && !$this->isRunning) {
-                            $this->isRunning = true;
-                            $this->dispatch(new WorkerRunningEvent($this));
-                            $this->handleTask($runner, $task);
+                        if (!$lockedTask->acquire() || $this->isRunning) {
```
[the remark this answers is not in the document]
Comment (jvancoillie) [reply src/Worker/Worker.php]: sorry for this late review, I hadn't clicked the button to send :(

[the remark this answers is not in the document]
Comment (Guikingone) [reply src/Worker/Worker.php]: Agree, could lead to something like this: 

```php
$lockedTask = $this->getLock($task);
if (!$lockedTask->acquire()) {
    continue;
}
```

Just before the `WorkerRunningEvent`.

[the remark this answers is not in the document]
Comment (jvancoillie) [reply src/Worker/Worker.php]: I like the idea, but I'm afraid that we will fall back into an infinite loop (there is no while break for this case) as we could have for `disabled|paused` tasks 😢

[the remark this answers is not in the document]
Comment (Guikingone) [reply src/Worker/Worker.php]: Yes, we're in a strange situation, I'm working on something 😅

[the remark this answers is not in the document]
Comment (Guikingone) [reply src/Worker/Worker.php]: I've pushed a fix, not sure about it for tonight, if it doesn't work, I will continue investigating tomorrow 🙂

[the remark this answers is not in the document]
Comment (jvancoillie) [reply src/Worker/Worker.php]: I tested the c6b3da9 commit ... It looks like it works 👍

[the remark this answers is not in the document]
Comment (Guikingone) [reply src/Worker/Worker.php]: If the latest commit fix the issue, I'm gonna fix Rector errors and merge this PR tomorrow 🙂

**Pull request merged**